### DISCUSSION TITLE
Safe to upgrade from 0.8.1 to 2.1.7?

### DISCUSSION THREAD

**Discussion by AGI-chandler** [openzfs/zfs#14396] [Q&A]
I obviously haven't upgraded zfs for while, but I usually don't upgrade things if they're working.  However, the reason this has been stuck on 0.8.1 for so long is that I had to manually compile it to use with our Intel QAT.  Well, I noticed a problem with the QAT drivers so I'm upgrading them to the latest version and thought I'd take the time to upgrade zfs as well.

I see a few other discussions on reddit about this sudden upgrade path along with issues that have happened.  I don't think any of those situations apply here.  The biggest issue seems to happen if a pool is encrypted? but we aren't using that, just compression, offloaded to the QAT device.  Seems I should be ok but it doesn't hurt to double check!  This particular pool is nearly 300TB compressed (nearly 2x), all backed up of course, but I'd rather not have to restore it from a backup!
hmm ~600TB@1Gbps would take over 61 Days! No thank you!

**Comment by m-ueberall** [comment]
I'd suggest to set up a separate test system to check whether the upgrade really works with your setup. Not sure whether this was explicitly mentioned on Reddit, but due to an ABI change, the names of the libraries changed (at least on Linux—if I'm not mistaken, this should be the case for all platforms?), which means that you'd need a separate boot medium/partition if you're using ZFS on root. As long as you don't activate any new features, it _should_ be possible to switch back (which is a scenario that I'd also check first using a test system given the size of your pool).

**Reply by rincebrain** [reply]
They changed on Linux, and FreeBSD wasn't supported until 2.0 so they changed there in a more drastic fashion, and everyone else isn't using this tree directly so they might have their own versioning.

But there's not really anything known to be on fire if you do this, at present, and if you find something, it'd be an important bug.

**Reply by AGI-chandler** [reply]
Thanks, well a test system would be nice... don't really have one at my disposal though.

Not using ZFS on root.

I got started early, figuring I could just go back to the old version, but seem to be [having a weird problem with the modules](https://github.com/openzfs/zfs/issues/14398), anyway, so I just might have to do that, maybe try 0.8.7 first...

and why would you give me a thumbs down, @scineram? lol
